server/src/main/java/io/crate/types/IntervalType.java

```diff
@@ -106,8 +105,33 @@ public Period sanitizeValue(Object value) {
 
     @Override
     public int compare(Period p1, Period p2) {
-        return toStandardDuration(p1).compareTo(toStandardDuration(p2));
-
+        var period1 = normalizeDaysMonthsYears(p1.normalizedStandard(PeriodType.yearMonthDayTime()));
+        var period2 = normalizeDaysMonthsYears(p2.normalizedStandard(PeriodType.yearMonthDayTime()));
+        var cmp = Integer.compare(period1.getYears(), period2.getYears());
```

Comment:
Would it improve performance if we avoided the intermediate Period instances?

Reply:
You mean in the new method? to avoid the `result` one, we can do, but the `normalizedStandard()` call creates a new period anyway, or do you mean something else?

Reply:
Or do you mean to precalculate days/months/years and only set them once at the end?

Reply:
Do you mind taking another look?